### PR TITLE
feat: Add window drag area

### PR DESCRIPTION
**Pull request by ZeroX-DG**
Closes https://github.com/marc2332/freya/issues/262

### CODE REVIEW

File: crates/components/src/window_drag_area.rs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+///
+#[allow(non_snake_case)]
+#[component]
+pub fn WindowDragArea(
+    /// The inner children for the WindowDragArea
+    children: Element
+) -> Element {
+    let platform = use_platform();
+
+    let onmousedown = move |_| platform.drag_window();
```
Comment (marc2332) [inline crates/components/src/window_drag_area.rs:32]:
```suggestion
    let onmousedown = move |e: MouseEvent| {
        e.stop_propagation();
        platform.drag_window();
    };
```

**Review by marc2332 — CHANGES_REQUESTED** [review]
Just like I had it in my mind, I will try it asap. Just left a tiny comment

File: examples/window_drag_area.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                width: "100%",
+                height: "50%",
+                label {
+                    text_align: "center",
+                    margin: "100 0",
+                    "Use the top half to drag the window"
+                }
+            }
+        }
+    )
```
Comment (marc2332) [inline examples/window_drag_area.rs:41]:
```suggestion
    rsx!(
        rect {
            width: "100%",
            height: "100%",
            WindowDragArea {
                rect {
                    width: "100%",
                    height: "50%",
                    background: "rgb(28, 28, 28)",
                    color: "white",
                    main_align: "center",
                    label {
                        width: "100%",
                        text_align: "center",
                        "Drag Me!"
                    }
                }
            }
            rect {
                width: "100%",
                height: "50%",
                main_align: "center",
                label {
                    text_align: "center",
                    "Use the top half to drag the window"
                }
            }
        }
    )
```

**Review by marc2332 — CHANGES_REQUESTED** [review]

**Comment by marc2332** [comment]
Just a comment to clean up the example, it works perfectly!

**Review by marc2332 — APPROVED** [review]

**Comment by marc2332** [comment]
Just saw https://github.com/ZeroX-DG/raven ! You are crazy haha 🚀 

Let me know if you have any issue/doubt/missing feature and I will be happy to help 😄

**Comment by ZeroX-DG** [comment]
Haha thanks! You're doing a great job fixing my feature requests lightning fast haha.

**Pull request merged**